lib/src/intl_suggestors/intl_message_syntax.dart

```diff
@@ -142,6 +142,22 @@ class MessageSyntax {
   String nameForNode(StringLiteral body,
       {String? initialName, bool startAtZero = false}) {
     var messageText = body.toSource();
+    //Strings from constant fields will not pass below condition so initialName will be unchanged.
+    // Needing condition initialName.endsWith('_intlFunction') bcz when any node has addTestId we need to give that
+    // priority, and will not override the node name.
+    if (body is StringInterpolation && (initialName == null || initialName.endsWith('_intlFunction'))) {
```

Comment:
This works, but I think we could have better code structure by backing up a little bit. The _intlFunction name gets passed in from functionCall and functionDefinition, where they call functionNameFor ahead of this, and that's hard-coded to return the name with _intlFunction in it, or else use the test Id. 

If we can create a name from the strings parts of the interpolation, then that's better than either the test id or _intlFunction. So maybe we can not pass in the initialName in that case, and only call the functionNameFor in here as a failure case.

Reply:
Some thoughts on the feedback, Please suggest.
I have applied these conditions to cover the following cases :

- If some values/DOM don't have `testID` and have only Variable in interpolation with space like: `'${someVariable} '`, This is rare but still possible. To cover this case we may have the method name as `_intlFunction`.
- In getter calls, we are receiving node as `null` in `nameForNode`.

If we update the `functionNameFor` method and remove the `_intlFunction` name generation, we might have some method names as null.

Reply:
Obviously, if we do this, then we'd also need to change the callers to not pass the initialName. In this situation I don't think we ever need to fall back to intlFunction0 or similar.